src/rtf-plugins/fixture-managers/yarpplugin/YarpPluginFixture.cpp

```diff
@@ -24,15 +24,21 @@ using namespace yarp::dev;
 
 PREPARE_FIXTURE_PLUGIN(YarpPluginFixture)
 
-bool YarpPluginFixture::scanPlugins(ConstString name){
+bool YarpPluginFixture::scanPlugin(yarp::os::ConstString name, yarp::os::ConstString type){
     YarpPluginSelector selector;
     selector.scan();
     Bottle lst=selector.getSelectedPlugins();
     bool res=false;
     for (int i=0; i<lst.size(); i++) {
         Value& options = lst.get(i);
-        if(name == options.check("name",Value("untitled")).asString())
-            res=true;
+        if(type=="portmonitor"){
```

Comment:
I think that in order to make it easier to extend later, this could be something like this:

```c++
if(!type.empty()) {
    //...
} else {
    // ...
}
```